lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -77,10 +85,71 @@ VAR
     '}
     fCalibMJRatio: LREAL;
 
+    // Linked to beam rate
+    {attribute 'pytmc' := '
+        pv: BeamRateFromEpics
+        link: TPG:SYS0:1:DST04:RATE
+    '}
+    fFrequency: UINT;
+
+    // Pulse-by-pulse energy
+    {attribute 'pytmc' := '
+        pv: ResponMJ
+        io: i
+        field: EGU mJ
+    '}
+    fResponMJ: LREAL;
+
+    // Pulse-by-pulse energy buffer
+    {attribute 'pytmc' := '
+        pv: ResponMJ_BUFFER
+        io: i
+        field: EGU mJ
+    '}
+    fResponMJBuffer: ARRAY[1..1000] OF LREAL;
+
+    // Wattage
+    {attribute 'pytmc' := '
+        pv: ResponWattage
+        io: i
+        field: EGU W
+    '}
+    fResponWattage: LREAL;
+
+    // Background voltage
+    {attribute 'pytmc' := '
+        pv: Background
+        io: io
+        field: EGU V
+    '}
+    fBackground: LREAL;
+
+    // Boolean to trigger collection of background voltages
+    {attribute 'pytmc' := '
+        pv: BackgroundCollect
+        io: i
+    '}
+    fBackgroundCollect: BOOL;
+
+    {attribute 'pytmc' := '
+        pv: BackgroundCollectionTime
+        io: io
+        field: EGU s
+    '}
+    // Time in seconds to collect background voltages for
+    fBackgroundCollectionTime: UINT;
+    // Number of cycles to collect background voltages for
+    fBackgroundCollectionCycles: UDINT;
+    // Number of cycles that background voltages have already been collected
+    fBackgroundCollectionCyclesDone: UDINT;
```

Comment:
f is the prefix for floating points, but this is actually assigned as a UDINT and other fs are assigned as BOOL